docs-v2/pages/connect/components.mdx

```diff
@@ -0,0 +1,501 @@
+
+The first thing to do in order to run a component for your end users is to know
+about the component itself. To start with, you need to know the component's
+unique key. Using that key, you can retrieve the component's structure from the
+Pipedream API. See the [component
+structure](/components/api#component-structure) section in our docs for more
+details.
+
+As an example, the following API call will return the structure of the "List
+Commits" component for Gitlab:
+
+```text
+GET /v1/connect/components/gitlab-list-commits
+```
+
+The response will contain the component's structure, including its
+(human-understandable) name, version, and most importantly, the configuration
+options that the component accepts (also known as "props", which is an
+abbreviation of "properties"). Here's an example of the response for the
+component in the example above:
+
+```json
+{
+  "data": {
+    "name": "List Commits",
+    "version": "0.0.3",
+    "key": "gitlab-list-commits",
+    "configurable_props": [
+      {
+        "name": "gitlab",
+        "type": "app",
+        "app": "gitlab"
+      },
+      {
+        "name": "projectId",
+        "type": "integer",
+        "label": "Project ID",
+        "description": "The project ID, as displayed in the main project page",
+        "remoteOptions": true
+      },
+      {
+        "name": "refName",
+        "type": "string",
+        "label": "Branch Name",
+        "description": "The name of the branch",
+        "remoteOptions": true,
+        "optional": true
+      },
+      {
+        "name": "max",
+        "type": "integer",
+        "label": "Max Results",
+        "description": "Max number of results to return. Default value is `100`",
+        "optional": true,
+        "default": 100
+      }
+    ]
+  }
+}
+```
+
+Using this information, you can now drive the configuration of the component for
+your end users, as described in the next section.
+
+### Configuration
+
+Component execution on behalf of your end users requires a few preliminary
+steps, focused on getting the right input parameters (a.k.a.
+[props](/workflows/using-props)) to the component.
+
+Configuring each prop for a component usually involves an API call to our
+backend to retrieve the possible values, unless the values that a prop can take
+are static/free-form. The endpoint is accessible at:
+
+```text
+POST /v1/connect/components/configure
+```
+
+The reason for this back-and-forth is because the options for a prop are usually
+linked to a specific user's account, and the Pipedream API is the one in charge
+of contacting the third-party service using the user's credentials, retrieving
+the necessary data, and sending these options back as a response. Please not
+that this logic is not implemented by the API itself, but by the component's
+code; in particular, the `options` function of the corresponding prop.
+
+The initial configuration call must contain your user's account ID for the
```

Comment:
This payload contains a lot more than just the `authProvisionId` and `prop_name`, so as a naive user I'm a little confused when I see more params, especially things like `async_handle` and `external_user_id`. Maybe describe those params and change this to a list to walk people through each param, e.g.

> When you call the `/v1/connect/components/configure` endpoint, you must pass:
>  * The `id` of the component — the component's `key`
> * The [`external_user_id`](/connect/api#external-users)
> ...